libdnf/advisory/advisory.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/advisory/advisory.hpp"
+
+#include "libdnf/advisory/advisory_collection.hpp"
+#include "libdnf/advisory/advisory_reference.hpp"
+#include "libdnf/common/exception.hpp"
+#include "libdnf/rpm/solv_sack_impl.hpp"
+#include "libdnf/utils/utils_internal.hpp"
+
+#include <fmt/format.h>
+#include <solv/pool.h>
+
+namespace libdnf::advisory {
+
+Advisory::Advisory(libdnf::rpm::SolvSack & sack, AdvisoryId id) : id(id), sack(sack.get_weak_ptr()) {}
+
+std::string Advisory::get_name() const {
+    const char * name;
+    Pool * pool = sack->p_impl->get_pool();
+
+    name = pool_lookup_str(pool, id.id, SOLVABLE_NAME);
+
+    if (strncmp(
+            libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX, name, libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX_LENGTH) !=
+        0) {
+        auto msg = fmt::format(
+            R"**(Bad libsolv id for advisory "{}", solvable name "{}" doesn't have advisory prefix "{}")**",
+            id.id,
+            name,
+            libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX);
+        throw RuntimeError(msg);
+    }
+
+    return std::string(name + libdnf::utils::SOLVABLE_NAME_ADVISORY_PREFIX_LENGTH);
+}
+
+Advisory::Type Advisory::get_type() const {
```

Comment:
I like the method, but what about to add a method that will return an original value from libsolv (cstring or string)?

Reply:
Added.